kernel/arch/dreamcast/hardware/sq.c

```diff
@@ -72,6 +72,48 @@ void * sq_cpy(void *dest, const void *src, int n) {
     return dest;
 }
 
+/* copies n bytes from src to dest, dest must be 64-byte aligned */
+void* sq_cpy64(void *dest, const void *src, int n)
+{
+
+  uint32 *sq;
+  uint32 *d, *s;
+  
+  d = (uint32 *)(0xe0000000 | (((uint32)dest) & 0x03ffffe0));
+  s = (uint32 *)(src);
+  
+  
+  *((volatile unsigned int*)0xFF000038) = ((((uint32)dest)>>26)<<2)&0x1c;
+  *((volatile unsigned int*)0xFF00003C) = ((((uint32)dest)>>26)<<2)&0x1c;
+  
+  n >>= 6;
+  while (n--) 
```

Comment:
Note to self: BlueCrab commented on this algorithm never prefetching from the source buffer... This is something to play with when I am set up to profile with the perf counters.